Sources/SchafKit/Extensions/Collections/Array.swift

```diff
@@ -100,6 +100,50 @@ public extension Array {
         }
         return false
     }
+    
+    /// Get the max value of a field of the array element type.
+    ///
+    /// ```
+    /// ["foo", "bar", "toast"].max(by: \.count)
+    /// ```
+    ///
+    /// - Parameter f: A function/keypath returning the value to be compared.
+    func max<T: Comparable>(by f: (Element) -> T) -> T? {
```

Comment:
Please do not call variables of a public function "f" in Swift, especially if you document them. Same for all other proposed changes.

Reply:
Done